assets/src/edit-story/components/library/panes/shapes/shapePreview.js

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { createRef } from 'react';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import useLibrary from '../../useLibrary';
+import createSolid from '../../../../utils/createSolid';
+import { PAGE_WIDTH } from '../../../../constants';
+
+// By default, the element should be 33% of the page.
+const DEFAULT_ELEMENT_WIDTH = PAGE_WIDTH / 3;
+const PREVIEW_SIZE = 36;
+
+const ShapePreviewContainer = styled.div`
+  position: relative;
+  padding: 0.8em 0.5em;
+  flex: 0 0 25%;
+  display: flex;
+  justify-content: center;
+  cursor: pointer;
+`;
+
+const Path = styled.path`
+  fill: ${({ theme }) => theme.colors.fg.v1};
+`;
+
+function ShapePreview({ mask }) {
+  const {
+    actions: { insertElement },
+  } = useLibrary();
+
+  // Creating a ref to the Path so that it can be used as a drag icon.
+  // This avoids the drag image that follows the cursor from being the whole
+  // component with large paddings, and only drags the svg part of it.
+  const pathRef = createRef();
```

Comment:
It still has a gray background though - is that intended or? Would be much nicer if it was only the shape.

Reply:
I mentioned this in the main issue and we agreed to fix that grew bug later. I'm not sure how to solve the grey background issue but I proposed a few solutions to it. Could you take a look and let me know what you think? #1579

Reply:
The gray background is merely the default color set in the shape.

We should probably tag in @samitron7 to decide on this - is it okay that you drag in a white shape and it turns gray on drop?

Reply:
I have no idea how to solve that. If @dvoytenko and @wassgha can't help you, neither can I 😉

Reply:
Should we just create a new issue for the grey background? I think the dragging functionality itself should be sorted out first since that's a bit of a complicated issue.

Reply:
It's actually partially covered by the new issue #1930, so we can probably punt this to that one.